lib/monitoring/aws-cloudwatch/SyntheticsCanaryMetricFactory.ts

```diff
@@ -0,0 +1,103 @@
+import { DimensionHash } from "monocdk/aws-cloudwatch";
+import { Canary } from "monocdk/aws-synthetics";
+import {
+  MetricFactory,
+  MetricStatistic,
+  RateComputationMethod,
+} from "../../common";
+
+const MetricNamespace = "CloudWatchSynthetics";
+
+export interface SyntheticsCanaryMetricFactoryProps {
+  /**
+   * CloudWatch Canary to monitor
+   */
+  readonly canary: Canary;
+  /**
+   * Method used to calculate relative rates
+   * @default average
+   */
+  readonly rateComputationMethod?: RateComputationMethod;
+}
+
+export class SyntheticsCanaryMetricFactory {
+  protected readonly metricFactory: MetricFactory;
+  protected readonly rateComputationMethod: RateComputationMethod;
+  protected readonly dimensions: DimensionHash;
+
+  constructor(
+    metricFactory: MetricFactory,
+    props: SyntheticsCanaryMetricFactoryProps
+  ) {
+    this.metricFactory = metricFactory;
+    this.rateComputationMethod =
+      props.rateComputationMethod ?? RateComputationMethod.AVERAGE;
+    this.dimensions = {
+      CanaryName: props.canary.canaryName,
+    };
+  }
+
+  metricLatencyAverageInMillis() {
```

Comment:
It seems like some of these metrics are already abstracted as L2 functions: https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-synthetics.Canary.html#metricwbrdurationoptionsspan-classapi-icon-api-icon-experimental-titlethis-api-element-is-experimental-it-may-change-without-noticespan

Reply:
I can use them. I just had a bad experience once (I think with DDB metrics) when they got broken, or used invalid statistic. That is why I made them from scratch.

Reply:
It's probably fair to rely on AWS CDK being fixed rather than us being responsible for having to patch any issues.

Reply:
Publishing updated commit.